hedera-node/src/main/java/com/hedera/services/state/logic/NetworkCtxManager.java

```diff
@@ -125,9 +129,11 @@ public void advanceConsensusClockTo(Instant consensusTime) {
 
 			/* We only check whether the midnight rates should be updated every intervalSecs in consensus time */
 			if (elapsedInterval >= intervalSecs) {
```

Comment:
I'm not sure what the value of `ratesMidnightCheckInterval ` is, but it could perhaps cause the stake info to not occur for the first transaction after midnight UTC and instead occur some period after. At minimum, it's confusing to use a property referring to exchange rates.

Reply:
I think we should rename this property. `ratesMidnightCheckInterval` gives us how often should we be checking if we have to do perform `end of staking period calculations` / `update midnight rates`. Since it was only pertaining to exchange rate - midnight rate before we named the property that way.

Reply:
But moving it to a const doesn't help if it violates the invariant that nodestakeupdate should be directly after the first transaction after midnight, not 1s after midnight.